ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/visitor/infraestructure/antlr/visitors/JavaMethodVisitor.java

```diff
@@ -0,0 +1,155 @@
+                        dependencyAnalyzer.analyze(methodCtx),
+                new DependencyInfo(new ArrayList<>(), new ArrayList<>(), new ArrayList<>())
+        );
+    }
+
+    @Override
+    public List<String> getMethodModifiers(ParserRuleContext ctx) {
+        return validateAndExecute(ctx, methodCtx -> {
+            List<String> modifiers = new ArrayList<>();
+
+            ParserRuleContext parent = methodCtx.getParent();
+            if (parent instanceof JavaParser.MemberDeclarationContext) {
+                parent = parent.getParent();
+            }
+
+            if (parent instanceof JavaParser.ClassBodyDeclarationContext parentCtx) {
+                List<JavaParser.ModifierContext> modifierContexts = parentCtx.modifier();
+                if (modifierContexts != null) {
+                    for (JavaParser.ModifierContext mod : modifierContexts) {
+                        if (mod.classOrInterfaceModifier() != null) {
+                            if (mod.classOrInterfaceModifier().PUBLIC() != null) {
+                                modifiers.add("public");
+                            } else if (mod.classOrInterfaceModifier().PRIVATE() != null) {
+                                modifiers.add("private");
+                            } else if (mod.classOrInterfaceModifier().PROTECTED() != null) {
+                                modifiers.add("protected");
+                            } else if (mod.classOrInterfaceModifier().STATIC() != null) {
+                                modifiers.add("static");
+                            } else if (mod.classOrInterfaceModifier().FINAL() != null) {
+                                modifiers.add("final");
+                            }
+                        }
+                    }
+                }
+            }
+            return modifiers;
+        }, new ArrayList<>());
+    }
+
+    @Override
+    public String getReturnType(ParserRuleContext ctx) {
+        return validateAndExecute(ctx, methodCtx ->
+            methodCtx.typeTypeOrVoid().getText(), ""
+        );
+    }
+
+    @Override
+    public List<ParameterInfo> getParameters(ParserRuleContext ctx) {
+        return validateAndExecute(ctx, methodCtx -> {
+            List<ParameterInfo> parameters = new ArrayList<>();
+            if (methodCtx.formalParameters().formalParameterList() != null) {
+                methodCtx.formalParameters()
+                        .formalParameterList()
+                        .formalParameter()
+                        .forEach(param -> {
+                            String paramName = param.variableDeclaratorId().getText();
+                            String paramType = param.typeType().getText();
+                            parameters.add(new ParameterInfo(paramName, paramType));
+                        });
+            }
+
+            return parameters;
+        }, new ArrayList<>());
+    }
+
+    private <T> T validateAndExecute(
```

Comment:
Take it to another class so that you can then call it in this class, since the same method will also be used in other classes.